src/timeseriesinsights/azext_timeseriesinsights/custom.py

```diff
@@ -0,0 +1,229 @@
+        tags=tags,
+        sku=Sku(name=sku_name, capacity=sku_capacity) if sku_name and sku_capacity else None,
+        data_retention_time=data_retention_time,
+        storage_limit_exceeded_behavior=storage_limit_exceeded_behavior,
+        partition_key_properties=[TimeSeriesIdProperty(name=id_property, type="String") for id_property in partition_key_properties] if partition_key_properties else None
+    )
+    return client.update(resource_group_name=resource_group_name, environment_name=environment_name, parameters=parameters)
+
+
+def create_timeseriesinsights_environment_longterm(cmd, client,
+                                                   resource_group_name, environment_name,
+                                                   sku_name, sku_capacity,
+                                                   time_series_id_properties,
+                                                   storage_account_name,
+                                                   storage_management_key,
+                                                   data_retention,
+                                                   location=None, tags=None):
+    from .vendored_sdks.timeseriesinsights.models import LongTermEnvironmentCreateOrUpdateParameters, Sku, \
+        TimeSeriesIdProperty, LongTermStorageConfigurationInput
+    parameters = LongTermEnvironmentCreateOrUpdateParameters(
+        location=location,
+        tags=tags,
+        sku=Sku(name=sku_name, capacity=sku_capacity),
+        time_series_id_properties=[TimeSeriesIdProperty(name=time_series_id_properties, type="String")],
+        storage_configuration=LongTermStorageConfigurationInput(account_name=storage_account_name, management_key=storage_management_key),
+        data_retention=data_retention)
+    return client.create_or_update(resource_group_name=resource_group_name, environment_name=environment_name, parameters=parameters)
+
+
+def update_timeseriesinsights_environment_longterm(cmd, client,
+                                                   resource_group_name, environment_name,
+                                                   storage_management_key=None,
+                                                   data_retention=None,
+                                                   tags=None):
+    from .vendored_sdks.timeseriesinsights.models import LongTermEnvironmentUpdateParameters, \
+        LongTermStorageConfigurationMutableProperties
+    parameters = LongTermEnvironmentUpdateParameters(
+        tags=tags,
+        storage_configuration=LongTermStorageConfigurationMutableProperties(management_key=storage_management_key) if storage_management_key else None,
+        data_retention=data_retention)
+    return client.update(resource_group_name=resource_group_name, environment_name=environment_name, parameters=parameters)
+
+
+def list_timeseriesinsights_environment(cmd, client, resource_group_name=None):
+    if resource_group_name is not None:
```

Comment:
Use if resource_group_name: in order to use the workaround -g= to temporarily disable appending default resource group

Reply:
Nice catch!